server/internal/provider/kaetzchen/cbor_plugins.go

```diff
@@ -136,34 +136,33 @@ func (k *CBORPluginWorker) processKaetzchen(pkt *packet.Packet, pluginClient *cb
 		HasSURB: surb != nil,
 	}
 	cborResponse := <-pluginClient.ReadChan()
-	switch r := cborResponse.(type) {
-	case *cborplugin.Response:
-		if len(r.Payload) > k.geo.UserForwardPayloadLength {
-			// response is probably invalid, so drop it
-			k.log.Errorf("Got response too long: %d > max (%d)",
-				len(r.Payload), k.geo.UserForwardPayloadLength)
-			instrument.KaetzchenRequestsDropped(1)
+	payload, err = cborResponse.Marshal()
```

Comment:
Reusing `payload` here is slightly confusing. It would be less error-prone if we used separate variables for the request payload and the response payload.

Reply:
this got refactored elsewhere and merged from main.